### PR TITLE
#139: add examples to regexp in roll engine

### LINKED ISSUE #139: Roll Engine: Add in-line comment documentation for regex instances

### PR DESCRIPTION
**Pull request by ntseng**
Summary
-------
- add examples to regexp in roll engine

Local Tests Performed
---------------------
- [x] bot still turns on
- documentation only

Issue
-----
Closes #139

### CODE REVIEW

File: source/engines/rollEngine.js [hunk 1]
```diff
@@ -161,24 +161,24 @@ class ResultSet {
...
 			return new DieSelectResultSet(parsingString.slice(0, dPos), parsingString.slice(dPos - parsingString.length, selPos), parsingString.slice(selPos - parsingString.length, endNumPos), parsingString.slice(endNumPos, parsingString.length))
-		} else if (/^\d+d(\%|\d+)$/.test(parsingString)) { //parse multiple die roll
+		} else if (/^\d+d(\%|\d+)$/.test(parsingString)) { //parse multiple die roll // 1d20 + 1d6
 			var dPos = parsingString.search('d');
 			return new MultiDieResultSet(parsingString.slice(0, dPos), parsingString.slice(dPos - parsingString.length));
-		} else if (/^-?\d+$/.test(parsingString)) { //parse number result
+		} else if (/^-?\d+$/.test(parsingString)) { //parse number result // 20, -3
 			return new SingleResultSet(parsingString, false);
-		} else if (/^1?u?l\[.*\]/.test(parsingString)) { //parse single list roll
+		} else if (/^1?u?l\[.*\]/.test(parsingString)) { //parse single list roll // 1l["a", "b", "c"], 1ul[a, b, c, d]
```
Comment (SillySalamndr) [inline source/engines/rollEngine.js]: Having one of these examples leave out the 1 would make it clearer that it is technically optional for this form.

File: source/engines/rollEngine.js [hunk 2]
```diff
@@ -161,24 +161,24 @@ class ResultSet {
 		// Handle base cases
 		if (/^1?d(\%|\d+)$/.test(parsingString)) { //parse single die roll //(\%|\d+) -> d%, 1d5, d75
 			return new SingleResultSet(parsingString);
-		} else if (/^\d+d(\%|\d+)[dk][lh]?\d+$/.test(parsingString)) { //parse multiple die with selection
+		} else if (/^\d+d(\%|\d+)[dk][lh]?\d+$/.test(parsingString)) { //parse multiple die with selection // 1d20dl, 2d4kh2
```
Comment (SillySalamndr) [inline source/engines/rollEngine.js]: Adding an example for drop lowest would cover cases better

File: source/engines/rollEngine.js [hunk 3]
```diff
@@ -161,24 +161,24 @@ class ResultSet {
...
 		if (/^1?d(\%|\d+)$/.test(parsingString)) { //parse single die roll //(\%|\d+) -> d%, 1d5, d75
 			return new SingleResultSet(parsingString);
-		} else if (/^\d+d(\%|\d+)[dk][lh]?\d+$/.test(parsingString)) { //parse multiple die with selection
+		} else if (/^\d+d(\%|\d+)[dk][lh]?\d+$/.test(parsingString)) { //parse multiple die with selection // 1d20dl, 2d4kh2
 			var dPos = parsingString.search('d');
 			var selPos = parsingString.search(/[dk][lh]?\d+$/);
 			var endNumPos = parsingString.search(/\d+$/);
 			return new DieSelectResultSet(parsingString.slice(0, dPos), parsingString.slice(dPos - parsingString.length, selPos), parsingString.slice(selPos - parsingString.length, endNumPos), parsingString.slice(endNumPos, parsingString.length))
-		} else if (/^\d+d(\%|\d+)$/.test(parsingString)) { //parse multiple die roll
+		} else if (/^\d+d(\%|\d+)$/.test(parsingString)) { //parse multiple die roll // 1d20 + 1d6
```
Comment (SillySalamndr) [inline source/engines/rollEngine.js]: This case is not for addition, but is instead for rolling one or more dice of the same size. (Addition is handled by cases not found in this review.) This also means that the example(s) for this should include one that rolls more than one die of that size.

**Comment by SillySalamndr** [comment]
Do we want to add examples for the cases from lines 197 to 209? Should that be a different task?

**Comment by ntseng** [comment]
Whoops, missed those. I'll add them now.

File: source/engines/rollEngine.js (hunk 3 above)
Comment (SillySalamndr) [reply source/engines/rollEngine.js]: We really should fix this one, even if we decide against the other two.

**Review by SillySalamndr — CHANGES_REQUESTED** [review]
There's at least one comment on one of the comments (geez, that'll get confusing) that should be fixed, since it will cause confusion down the line. Everything else is good enough for our cases.

**Review by SillySalamndr — APPROVED** [review]

**Pull request merged**